app/models/playlist.js

```diff
@@ -90,9 +90,29 @@ const Playlist = Backbone.Model.extend({
     addCurrentPlaylistToUpNext() {
         let currentPlaylist = application.appState.get('currentPlaylist');
         let currentTracks = currentPlaylist.get('tracks');
-        currentTracks.each(track => {
-            this.addTrack(track);
-        });
+        let sort = application.appState.get('sort');
+        if (sort.by != 'default') {
+            if (sort.direction == 'reverse') {
+                let reverse = sort.by.last(sort.by.length).reverse();
+                _.each(reverse, track => {
+                    let t = currentTracks.get(track.get('_id'))
+                    if (t) {
+                        this.addTrack(t);
+                    }
+                });
+            } else {
+                sort.by.each(track => {
+                    let t = currentTracks.get(track.get('_id'))
+                    if (t) {
+                        this.addTrack(t);
+                    }
+                });
+            }
```

Comment:
@flyingrub Don't you think we can share some code between the `if` and `else` banches? Maybe something like:

```js
let tracksToAdd = sort.by;
if (sort.direction == 'reverse') {
    tracksToAdd = sort.by.last(sort.by.length).reverse();
}
_.each(reverse, track => {
    let t = currentTracks.get(track.get('_id'))
    if (t) {
         this.addTrack(t);
    }
});
```